### PR TITLE
[ITEM-295] Agent stats per queues

### PR DESCRIPTION
**Pull request by rbienvenu0**
- **dao: add queue_id to agent stats**
- **stats: link agent stats to queues**

---

> [!NOTE]
> **Medium Risk**
> Changes how reported agent login/pause/wrapup time is computed and stored; incorrect interval logic would skew operational stats, but scope is stats aggregation rather than call handling or auth.
> 
> **Overview**
> Adds **per-queue agent periodic statistics** alongside existing agent-wide totals.
> 
> `stat_agent_periodic` gains optional `stat_queue_id` (FK + index). DAOs now key metrics by `(agent_id, queue_id_or_name)` instead of agent alone: `insert_stats` writes separate global (`None`) and per-queue rows; `get_wrapup_times` includes queue name and rolls wrapup into both global and queue keys (deduping when `queuename` is NULL).
> 
> `stat_dao` extends login/pause interval logic from queue_log: **ADDMEMBER/REMOVEMEMBER** membership intervals intersect global login time for per-queue login; **PAUSE/UNPAUSE** per queue plus **PAUSEALL** intervals intersected with membership for per-queue pause. Shared `_get_open_close_intervals_by_queue` resolves agents via name or `Local/id-*@agentcallback` interface. NULL `queuename` events are skipped for per-queue keys.
> 
> Tests updated for tuple keys and broad coverage of edge cases (open-at-range-start, logoff clipping, interface vs membername).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7f11bcb479ca8c8a61ae4a41fd091dfb493257f1. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: xivo_dao/stat_dao.py [hunk 1]
```diff
@@ -211,13 +211,58 @@ def get_login_intervals_in_range(session, start, end):
...
+) membership_events
+WINDOW w AS (PARTITION BY agent_id, queue_name ORDER BY event_time)
+ORDER BY agent_id, queue_name, event_time
+'''
+    formatted_start = start.strftime(_STR_TIME_FMT)
+    formatted_end = end.strftime(_STR_TIME_FMT)
+    rows = session.execute(
+        text(query),
+        {'start': formatted_start, 'end': formatted_end},
+    )
```
[the remark this answers is not in the document]
Comment (rbienvenu0) [reply xivo_dao/stat_dao.py:308]: Updated

File: xivo_dao/queue_log_dao.py [hunk 2]
```diff
@@ -40,15 +41,24 @@ def get_wrapup_times(session, start, end, interval):
...
     results = {}
     for row in rows.all():
-        agent_id, wstart, wend = row.agent_id, row.start, row.end
+        agent_id, queue_name, wstart, wend = (
+            row.agent_id,
+            row.queue_name,
+            row.start,
+            row.end,
+        )
+        keys = ((agent_id, None), (agent_id, queue_name))
```
[the remark this answers is not in the document]
Comment (rbienvenu0) [reply xivo_dao/queue_log_dao.py]: It's a false positive, the resolution happens in wazo_stat's `_resolve_queue_names`

**Review by bloom1 — APPROVED** [review]
